### PR TITLE
Add highlights model

### LINKED ISSUE #220: Add Highlights model
Related to https://github.com/mozilla/network/issues/203

Schema: https://github.com/mozilla/network/blob/master/source/json/highlights.schema.json
Current static JSON: https://github.com/mozilla/network/blob/master/source/json/highlights.json

### PR DESCRIPTION
**Pull request by alanmoo**
Closes #220

### CODE REVIEW

File: app/networkapi/templates/highlights/adminsortable_objects_custom.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{% load adminsortable_tags admin_urls staticfiles %}
```
Comment (alanmoo) [inline app/networkapi/templates/highlights/adminsortable_objects_custom.html]: This is a copy/paste job from "people". I want to put this somewhere common, but I'm not sure I can.

Comment (Pomax) [inline app/networkapi/templates/highlights/adminsortable_objects_custom.html]: yeah I don't think `{% %}` can be offloaded to secondary files =(

Comment (alanmoo) [inline app/networkapi/templates/highlights/adminsortable_objects_custom.html]: I meant the whole file

File: app/networkapi/settings.py [hunk 2]
```diff
@@ -253,21 +255,15 @@
 ADMIN_MENU_ORDER = (
     ('Content', ('pages.Page',
                  ('Media Library', 'media-library'))),
-    ('Data', ('people.Person', 'news.News', 'people.InternetHealthIssue',)),
+    ('Data', (
+        'people.Person',
+        'news.News',
+        'people.InternetHealthIssue',
+        'highlights.Highlight',
+        )),
```
Comment (Pomax) [inline app/networkapi/settings.py]: damn, flake8 accepts this?

File: app/networkapi/templates/highlights/adminsortable_objects_custom.html (hunk 1 above)
Comment (Pomax) [inline app/networkapi/templates/highlights/adminsortable_objects_custom.html]: Turn the "this file" specific code into a `{% block %}`, and put the rest in a separte file, then have this file `{% extends ... %}`, and that should do the trick.

File: app/networkapi/highlights/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
```
Comment (arush0311) [inline app/networkapi/highlights/migrations/0001_initial.py]: Do we have existing features data in database?

File: app/networkapi/highlights/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            name='Highlight',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('title', models.CharField(help_text='Title of the higlight', max_length=300)),
+                ('description', models.TextField(help_text='Description of the higlight', max_length=5000)),
+                ('link_label', models.CharField(help_text="Text to show that links to this higlight's details page", max_length=300)),
+                ('link_url', models.URLField(help_text="Link to this higlight's details page", max_length=2048)),
+                ('image', models.FileField(help_text='Image representing this highlight', max_length=2048, upload_to=networkapi.highlights.models.get_highlights_image_upload_path)),
+                ('featured', models.BooleanField(default=False, help_text='Do you want to feature this highlight?')),
+                ('footer', mezzanine.core.fields.RichTextField(help_text='Content to appear after description (view more projects link or something similar)', null=True, verbose_name='footer')),
```
Comment (Pomax) [inline app/networkapi/highlights/migrations/0001_initial.py]: This is a dubious name: "footers" are things at the bottom of a page, whereas this refers to (near as I can figure) something that allows the user to click through to "something". Can we change the name for this field?

File: app/networkapi/highlights/models.py [hunk 5]
```diff
@@ -27,7 +27,12 @@ def published(self):
         )
 
 
-class Highlight(models.Model):
+class Highlight(SortableMixin):
+    """
+    An data type to highlight things like pulse
```
Comment (Pomax) [inline app/networkapi/highlights/models.py]: `A data type`?

File: app/networkapi/static/highlights/js/admin.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+function run($) {
+  var fieldsTiedToFeaturedCards = ".quote, .bio, .interview_url";
+
+  $(".featured").on('click', function() {
+    if(!$("#id_featured").is(':checked')) {
```
Comment (Pomax) [inline app/networkapi/static/highlights/js/admin.js]: can `$("#id_featured")` be cached so it's not constantly re-retrieved during click handling?

File: app/networkapi/settings.py (hunk 2 above)
Comment (alanmoo) [inline app/networkapi/settings.py]: ¯\_(ツ)_/¯ apparently

File: app/networkapi/highlights/migrations/0001_initial.py (hunk 3 above)
Comment (alanmoo) [inline app/networkapi/highlights/migrations/0001_initial.py]: No, though the model might exist in the database structure.

File: app/networkapi/highlights/migrations/0001_initial.py (hunk 4 above)
Comment (gvn) [inline app/networkapi/highlights/migrations/0001_initial.py]: `footer` is now an HTML element that can be used in multiple contexts.

What do you propose as an alternative?

Comment (alanmoo) [inline app/networkapi/highlights/migrations/0001_initial.py]: Agreed with @gvn, this is semantically fine and the help text explains it.

**Pull request merged**